babel/numbers.py

```diff
@@ -385,6 +384,205 @@ def format_scientific(number, format=None, locale=LC_NUMERIC):
+    u'1\\xa0200 tommer kvikks\\xf8lv'
+
+    Number formats may be overridden with the ``format`` parameter.
+
+    >>> format_unit(Decimal("-42.774"), 'temperature-celsius', 'short', format='#.0', locale='fr')
+    u'-42,8 \\xb0C'
+
+    The locale's usual pluralization rules are respected.
+
+    >>> format_unit(1, 'length-meter', locale='ro_RO')
+    u'1 metru'
+    >>> format_unit(0, 'length-picometer', locale='cy')
+    u'0 picometr'
+    >>> format_unit(2, 'length-picometer', locale='cy')
+    u'2 bicometr'
+    >>> format_unit(3, 'length-picometer', locale='cy')
+    u'3 phicometr'
+
+    >>> format_unit(15, 'length-horse', locale='fi')
+    Traceback (most recent call last):
+        ...
+    ValueError: length-horse/long is not a known unit/length in fi
+
+    .. versionadded:: 2.2.0
+
+    :param value: the value to format.
+    :param measurement_unit: the code of a measurement unit.
+                             Known units can be found in the CLDR Unit Validity XML file:
+                             http://unicode.org/repos/cldr/tags/latest/common/validity/unit.xml
+    :param length: "short", "long" or "narrow"
+    :param format: An optional format, as accepted by `format_decimal`.
+    :param locale: the `Locale` object or locale identifier
+    """
+    locale = Locale.parse(locale)
+
+    unit_length = _find_unit_pattern(measurement_unit, length, locale)
+    if not unit_length:
+        raise ValueError("%s is not a known unit with length %s in %s" % (measurement_unit, length, locale))
+    unit_patterns = locale.unit_patterns[unit_length]
+    formatted_value = format_decimal(value, format, locale)
+    plural_form = locale.plural_form(value)
+    if plural_form in unit_patterns:
+        return unit_patterns[plural_form].format(formatted_value)
+    # Fall back to a bad representation.
+    return '%s %s' % (formatted_value, measurement_unit)
+
+
+def _find_compound_unit(numerator_unit, denominator_unit, length='long', locale=LC_NUMERIC):
+    """
+    Find a predefined compound unit pattern.
+
+    Used internally by format_compound_unit.
+
+    >>> _find_compound_unit("kilometer", "hour", locale="en")
+    'speed-kilometer-per-hour:long'
+
+    >>> _find_compound_unit("mile", "gallon", locale="en")
+    'consumption-mile-per-gallon:long'
+
+    :param numerator_unit: The numerator unit's identifier
+    :param denominator_unit: The denominator unit's identifier
+    :param length: The length for the resultant formatting.
+    :param locale: the `Locale` object or locale identifier
+    :return: A key to the `unit_patterns` mapping, or None.
+    """
+    locale = Locale.parse(locale)
+    numerator_unit_length = _find_unit_pattern(numerator_unit, length, locale)
+    denominator_unit_length = _find_unit_pattern(denominator_unit, length, locale)
+    if not (numerator_unit_length and denominator_unit_length):
+        return None
+    # Since compound units are named "speed-kilometer-per-hour:long", we'll have to slice off
```

Comment:
Sure. That'll have to wait until 3.0 though, since someone might be using the raw data API.